chia-wallet/src/derive_synthetic.rs

```diff
@@ -0,0 +1,125 @@
+use chia_bls::{PublicKey, SecretKey};
+use num_bigint::BigInt;
+use sha2::{digest::FixedOutput, Digest, Sha256};
+
+pub trait DeriveSynthetic {
```

Comment:
is synthetic derivation definitely a wallet feature?
I'm not suggesting it belongs in `chia-bls`, I just want to point out that both hardened- and unhardened derivation is implemented there.

Reply:
The only use of it is in the standard transaction. Though given that it derives keys, I'm fine moving into chia-bls, don't really have an opinion either way.